### PR TITLE
Add docs for SimulatedCallingAssembly

### LINKED ISSUE #1354: PlatformNotSupportedException was thrown in an asp.net core app
Repro (nightly build .NET 6):

[WebTest.zip](https://github.com/dotnet/runtimelab/files/6871874/WebTest.zip)

```
dotnet publish -c Release -r win-x64
```

It's an asp.net core app created with default template. 

```
Unhandled Exception: System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Reflection.Assembly.GetCallingAssembly() + 0x5e
   at <Program>$.<Main>$(String[]) + 0x1d
   at WebTest!<BaseAddress>+0x9d8cdf
```

It turns out to be `GetCallingAssembly` was not supported in NativeAOT. Is there any solution or workaround to it? Or can `GetCallingAssembly` be made the same with `GetExecutingAssembly` (i.e. assuming all assemblies are being inlined)?

https://github.com/dotnet/aspnetcore/blob/0c5456afe95096db6ac5400cecd3440299b42714/src/DefaultBuilder/src/WebApplication.cs#L89
https://github.com/dotnet/aspnetcore/blob/0c5456afe95096db6ac5400cecd3440299b42714/src/DefaultBuilder/src/WebApplication.cs#L97

UPDATE: I found `Switch.System.Reflection.Assembly.SimulatedCallingAssembly` in source code, I think it should be added to using-nativeaot docs.

### PR DESCRIPTION
**Pull request by hez2010**
Fixes #1354

Though I think this should be made as default behavior.

### CODE REVIEW

File: docs/using-nativeaot/reflection-in-aot-mode.md [hunk 1]
```diff
@@ -69,10 +69,23 @@ In CoreRT, reflection metadata (names of types, list of their methods, fields, s
...
+
+If `Assembly.GetCallingAssembly` cannot be avoided, it can work with simulated calling assembly, which will return the entry assembly, i.e. the same with `Assembly.GetEntryAssembly`.
+
+To enable simulated calling assembly, you will need:
+
+```xml
+  <ItemGroup>
+    <RuntimeHostConfigurationOption Include="Switch.System.Reflection.Assembly.SimulatedCallingAssembly" Value="true" />
+  </ItemGroup>
+```
```
Comment (jkotas) [inline docs/using-nativeaot/reflection-in-aot-mode.md]:
```suggestion
## Shimming

Native AOT libraries have configuration settings (shims) that enable replacing some of frequently used reflection patterns that are incompatible with Native AOT with compatible equivalents that approximate their functionality, without changing the source code. The shim settings documented in this section are meant to be used as temporary unreliable workarounds until the permanent source code fix can be made. They are not guaranteed to make the application work correctly.

### Simulated calling assembly

`Assembly.GetCallingAssembly` is not supported in Native AOT and throws `PlatformNotSupportedException` by default. 

`Assembly.GetCallingAssembly` can be in certain situations simulated by `Assembly.GetEntryAssembly`. 

To enable simulated `Assembly.GetEntryAssembly`, you will need:
```

**Comment by jkotas** [comment]
> Though I think this should be made as default behavior.

We do want `GetCallingAssembly` to throw by default in NativeAOT . There are number of cases where the fallback behavior of `GetCallingAssembly` does not work well. We do not want to spend days trying to debug and diagnose those cases (we have been there with .NET Native for UWP).

File: docs/using-nativeaot/reflection-in-aot-mode.md [hunk 2]
```diff
@@ -69,10 +69,22 @@ In CoreRT, reflection metadata (names of types, list of their methods, fields, s
...
+
+Native AOT libraries have configuration settings (shims) that enable replacing some of frequently used reflection patterns that are incompatible with Native AOT with compatible equivalents that approximate their functionality, without changing the source code. The shim settings documented in this section are meant to be used as temporary unreliable workarounds until the permanent source code fix can be made. They are not guaranteed to make the application work correctly.
+
+### Simulated calling assembly
+
+`Assembly.GetCallingAssembly` is not supported in Native AOT and throws `PlatformNotSupportedException` by default. 
+
+`Assembly.GetCallingAssembly` can be in certain situations simulated by `Assembly.GetEntryAssembly`. 
+
+To enable simulated `Assembly.GetEntryAssembly`, you will need:
```
Comment (jkotas) [inline docs/using-nativeaot/reflection-in-aot-mode.md]:
```suggestion
To enable simulated `Assembly.GetEntryAssembly`, you will need:

```xml
  <ItemGroup>
    <RuntimeHostConfigurationOption Include="Switch.System.Reflection.Assembly.SimulatedCallingAssembly" Value="true" />
  </ItemGroup>
```
```

File: docs/using-nativeaot/reflection-in-aot-mode.md [hunk 3]
```diff
@@ -69,10 +69,27 @@ In CoreRT, reflection metadata (names of types, list of their methods, fields, s
...
+
```
Comment (jkotas) [inline docs/using-nativeaot/reflection-in-aot-mode.md]:
```suggestion

To enable simulated `Assembly.GetCallingAssembly`, you will need:

```

File: docs/using-nativeaot/reflection-in-aot-mode.md [hunk 4]
```diff
@@ -69,10 +69,27 @@ In CoreRT, reflection metadata (names of types, list of their methods, fields, s
...
+```xml
 <ItemGroup>
   <RdXmlFile Include="rd.xml" />
 </ItemGroup>
 ```
 
 Format of the file described [here](rd-xml-format.md)
+
+## Shimming
+
```
Comment (jkotas) [inline docs/using-nativeaot/reflection-in-aot-mode.md:81]: We have a few more of these shim settings. They can be documented in separate PRs.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!
